### PR TITLE
Feature/obligation validation

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**
Sets up the database for capture of obligation uploads and saving of this along with errors to the database. Unit and integration tests included.

### CODE REVIEW

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/SubmitSchemeObligationsHandlerTests.cs [hunk 1]
```diff
@@ -56,5 +140,74 @@ public async Task HandleAsync_GivenRequest_CsvDataShouldBeValidated()
...
+            
+            A.CallTo(() => obligationUploadValidator.Validate(A<List<ObligationCsvUpload>>._))
+                .Returns(obligationErrors);
+            A.CallTo(() => commonDataAccess.FetchCompetentAuthority(A<CompetentAuthority>._)).Returns(authority);
+
+            //act
+            await handler.HandleAsync(request);
+
+            //assert
+            A.CallTo(() => obligationDataAccess.AddObligationUpload(A<UKCompetentAuthority>.That.Matches(a => a.Equals(authority)),
```
Comment (meglenapnovoselska) [inline src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/SubmitSchemeObligationsHandlerTests.cs:178]: is the csv functionality only working for the UK Authority?

**Review by meglenapnovoselska — APPROVED** [review]

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/SubmitSchemeObligationsHandlerTests.cs (hunk 1 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/SubmitSchemeObligationsHandlerTests.cs:178]: UKCompetentAuthority is the domain object that makes to the Lookup.CompetentAuthority table in the database

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/ObligationDataAccessUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            userId = Guid.NewGuid();
+            fixture = new Fixture();
+
+            A.CallTo(() => userContext.UserId).Returns(userId);
+
+            obligationDataAccess = new ObligationDataAccess(context, userContext, genericDataAccess);
+        }
+
+        [Fact]
+        public async Task AddObligationUpload_GivenUploadData_ObligationUploadShouldBeAddedToContext()
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/ObligationDataAccessUnitTests.cs]: spit the tests out into two tests

File: src/EA.Weee.Integration.Tests/Handlers/SubmitSchemeObligationHandlerTests.cs [hunk 3]
```diff
@@ -1,63 +1,205 @@
...
+                {scheme2.ApprovalNumber},{scheme2.SchemeName},14,15,16,17,18,19,20,21,22,23,24,25,26,";
 
                 var fileInfo = new FileInfo("file name", Encoding.UTF8.GetBytes(csvHeader));
-                request = new SubmitSchemeObligation(fileInfo);
+                request = new SubmitSchemeObligation(fileInfo, CompetentAuthority.England);
             };
 
             private readonly Because of = () =>
             {
                 result = Task.Run(async () => await handler.HandleAsync(request)).Result;
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Integration.Tests/Handlers/SubmitSchemeObligationHandlerTests.cs:55]: Added, goood spot...

File: src/EA.Weee.DataAccess.Tests.Integration/DataAccess/ObligationDataAccessIntegration.cs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    using RequestHandlers.Shared;
+    using Weee.DataAccess.DataAccess;
+    using Weee.Tests.Core.Model;
+    using Xunit;
+    using CompetentAuthority = Core.Shared.CompetentAuthority;
+
+    public class ObligationDataAccessIntegration
+    {
+        [Fact]
+        public async Task AddObligationUpload_WithNoErrors_ShouldAddObligationUpload()
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.DataAccess.Tests.Integration/DataAccess/ObligationDataAccessIntegration.cs:25]: done

**Pull request merged**